crates/storage/src/index.rs

```diff
@@ -33,6 +33,12 @@ struct EntityIndex {
     /// Merkle hash of the entity's immediate data only. This gets combined with
     /// the hashes of its children to form the full hash.
     own_hash: [u8; 32],
+
+    /// Type identifier of the entity. This is noted so that the entity can be
+    /// deserialised correctly in the absence of other semantic information. It
+    /// is intended that the [`Path`](crate::address::Path) will be used to help
+    /// with this at some point, but at present paths are not fully utilised.
```

Comment:
I was going to mention that, paths are almost entirely useless atm, `find_by_path` is still unimplemented and it's unclear how to use paths to do anything

Reply:
You are correct. Paths are a fundamental part of the design, and important to certain aspects of the design, but as yet there are certain key questions about their usage that are outstanding, to be resolved. As such, given the fact that we've decided to go with a primarily ID-based approach in the first instance, the paths are sat waiting in the wings somewhat.